src/backend/distributed/utils/ruleutils_95.c

```diff
@@ -3624,7 +3624,7 @@ get_variable(Var *var, int levelsup, bool istoplevel, deparse_context *context)
 	else
 	{
 		/* System column - name is fixed, get it from the catalog */
-		attname = get_rte_attribute_name(rte, attnum);
+		attname = get_relid_attribute_name(rte->relid, attnum);
```

Comment:
There's a function `GetRangeTblKind` that you can call on the `rte` and it'll return `CITUS_RTE_SHARD` (or possibly `CITUS_RTE_REMOTE_QUERY`) for sharded relations. In that case, call `get_relid_attribute_name`, and leave the `else` block as it was.